### PR TITLE
Document conditional configuration

### LINKED ISSUE #180: Document that some plugins and rules are only available in certain environments
For example, the Web preset only exposes a minify plugin during `production`, leading to issues when trying to modify its settings, but blows up during `development`. Document that these configs should be modified conditionally based on the environment.

### PR DESCRIPTION
**Pull request by helfi92**
Resolves #180

### CODE REVIEW

File: docs/customization/advanced.md [hunk 1]
```diff
@@ -221,3 +221,17 @@ module.exports = neutrino => {
 Presets can also have their own custom data in addition to the Neutrino config. See your respective preset for details.
 Again, rather than reiterate the documentation for [webpack-chain](https://github.com/mozilla-neutrino/webpack-chain)
 here, please refer to its documentation for all ways you can modify a config instance to solve your use cases.
+
+### Conditional configuration
+
+Some plugins and rules are only available in certain environments. For example, the Web preset only exposes a minify
+plugin during production, leading to issues when trying to modify its settings, but blows up during development.
```
Comment (eliperelman) [inline docs/customization/advanced.md]: Switch "blows up" to "throws an exception". I typically avoid using colloquial prose in formal documentation.

**Comment by eliperelman** [comment]
Note that this won't completely solve #180. We need to walk through the documentation of all middleware and in the list of plugins, etc., mark if they are in all environments or in certain environments.

**Comment by helfi92** [comment]
@eliperelman makes sense. We only need to document the presets since they are the ones using the middlewares conditionally right? I made some [changes](https://github.com/mozilla-neutrino/neutrino-dev/pull/250/commits/6a589714dea2527f570cb9b4afc6ba2feedb39e2) to the documentation of the Web preset. I am planning to do the same thing for the rest of the presets. Let me know if I missed something or have a better idea.

**Comment by eliperelman** [comment]
@helfi92 OOOOO I love the table idea! Yeah, go ahead and update the presets, I'll take care of the middleware.

**Comment by helfi92** [comment]
Changed the Web and Node presets. I believe the rest don't add additional plugins.

**Comment by eliperelman** [comment]
@helfi92 looks like Jest and Mocha document their changes to the stuff they use.

**Pull request merged**